doc/manual/config-advanced.rst

```diff
@@ -78,10 +78,11 @@ Squid configuration for this might look like::
 Self-registration
 `````````````````
 There is also a configuration option to allow teams to self-register with
-the system: ``registration_category_name``. When left empty, no self-registration
-is allowed; when filled with a category name, newly registered teams will
-be placed in this category. During registration, a team can specify their
-affiliation.
+the system: ``registration_category_names``. When left empty, no
```

Comment:
I wonder whether this approach is the right one. Here you have a unchecked string field where you enter the category names. Perhaps it's better to add a boolean field to the team_category DB table whether self registration is allowed for this particular category or not.

Reply:
I like this idea. That would also remove a lot of checks whether the list is actually valid

Reply:
I've changed it to use a boolean field in the team_category DB table, and rebased (incorporating 464410198477; [steps](https://github.com/DOMjudge/domjudge/compare/40a6cc58723da6c093a369a4c3dc0a68082f53c9...a3eca5b8b1ef51c298a0d5d6000734da5e3b1fd6)).

Some notes:

 - When multiple categories are allowed for self-registration, it is difficult to toggle self-registration on and off (need to go into each category and toggle registration). I thought about adding a master configuration option, "allow_registration", that can be set to false to disable self-registration. (An option with that name existed before commit 9126537e1b. There can be an awkward state where allow_registration is true, but none of the categories are set to allow registration.)

 - I'd like to keep the configuration checker, even though no validation is required any more, so that there is a clear indication of whether self-registration is enabled or disabled. This way, even if the options that control self-registration are moved or changed in the future (e.g. are no longer tied to the team categories), there is a reliable place to check the status. The configuration checker could also display a hint on how to enable self-registration ("go to the Team Categories page"), but I haven't done that.

 - I used `@Serializer\Exclude()` on the new `$allow_self_registration` property in webapp/src/Entity/TeamCategory.php (copied from `$visible`), but I'm not sure if that's correct.

 - I wasn't sure whether to use underscores or camel case for the property name. I used underscores to match recent examples (`$orig_input_filename' in b8e74c1fc53c, `$team_categories` in 4a87d53f34b6), but then in the Twig template it needs to be in camel case for the getter to be inferred. (Also not sure about the doc style for the getter and setter; I tried to match the existing code.)

Reply:
> I've changed it to use a boolean field in the team_category DB table, and rebased (incorporating [4644101](https://github.com/DOMjudge/domjudge/commit/46441019847719337c20a4fa74cd60d10830a474); [steps](https://github.com/DOMjudge/domjudge/compare/40a6cc58723da6c093a369a4c3dc0a68082f53c9...a3eca5b8b1ef51c298a0d5d6000734da5e3b1fd6)).
> 
> Some notes:
> 
> * When multiple categories are allowed for self-registration, it is difficult to toggle self-registration on and off (need to go into each category and toggle registration). I thought about adding a master configuration option, "allow_registration", that can be set to false to disable self-registration. (An option with that name existed before commit [9126537](https://github.com/DOMjudge/domjudge/commit/9126537e1bf325a8310364a293b3a22c94df4ae7). There can be an awkward state where allow_registration is true, but none of the categories are set to allow registration.)

Yeah I'd say if we have the below point it's fine without a master switch?

> * I'd like to keep the configuration checker, even though no validation is required any more, so that there is a clear indication of whether self-registration is enabled or disabled. This way, even if the options that control self-registration are moved or changed in the future (e.g. are no longer tied to the team categories), there is a reliable place to check the status. The configuration checker could also display a hint on how to enable self-registration ("go to the Team Categories page"), but I haven't done that.

Sounds good to keep this checker.

> * I used `@Serializer\Exclude()` on the new `$allow_self_registration` property in webapp/src/Entity/TeamCategory.php (copied from `$visible`), but I'm not sure if that's correct.

This annotation is used so it is not exposed in the API. We could expose it, but then we need to make sure it is only for nonstrict clients. I'm fine with leaving it excluded for now.

> * I wasn't sure whether to use underscores or camel case for the property name. I used underscores to match recent examples (`$orig_input_filename' in b8e74c1fc53c, `$team_categories` in [4a87d53](https://github.com/DOMjudge/domjudge/commit/4a87d53f34b6b6591e4da7943122f335fcb93275)), but then in the Twig template it needs to be in camel case for the getter to be inferred. (Also not sure about the doc style for the getter and setter; I tried to match the existing code.)

We should indeed standardise this at some point...